packages/xrpl/src/sugar/autofill.ts

```diff
@@ -88,6 +97,42 @@ async function autofill<T extends Transaction>(
   return Promise.all(promises).then(() => tx)
 }
 
+async function setNetworkID(client: Client, tx: Transaction): Promise<void> {
+  if (client.networkID === undefined) {
+    await setNetworkIDAndBuildVersion(client)
+  }
```

Comment:
Also, given that not having the `NetworkID` when you need it can cause all future transactions to fail - it might be worth adding retry logic if you send a `server_info` request at the start of `Client.connect()` and it doesn't get a response. (If it does get a response without a `NetworkID` - that just means it's talking to mainnet or a similar network and does not need to retry).

Reply:
I agree with this except that if server info fails i feel that you probably don't have a valid connection to the socket.  That should be the fastest call if not one of the fastest calls.

Reply:
Note: `server_info` will still return a `NetworkID` even if this feature hasn't been added to the network yet, and/or the `NetworkID` doesn't need to be included in the transaction (like with testnet).
You can see that with the current Testnet: https://xrpl.org/websocket-api-tool.html?server=wss%3A%2F%2Fs.altnet.rippletest.net%3A51233%2F&req=%7B%22id%22%3A1%2C%22command%22%3A%22server_info%22%7D